### ISSUE TITLE
feature: exclude/ignore paths via --glob flag

### ISSUE THREAD

**Issue by acalvo** [vale-cli/vale#254]
Tightly related with https://github.com/errata-ai/vale/issues/243

See this min repro: https://github.com/acalvo/vale-glob

It has the following structure:
```
- content/
  - a.md
  - b/
    - b.md
  - c.md
```

If I run `vale ./content` I get the following result:

```
 content/a.md
 1:1  error  Did you really mean 'Lorem'?  Vale.Spelling 
 1:7  error  Did you really mean 'ipsum'?  Vale.Spelling 


 content/b/b.md
 1:11  error  Did you really mean 'amet'?  Vale.Spelling 


 content/c.md
 1:1  error  Did you really mean 'Lorem'?  Vale.Spelling 
 1:7  error  Did you really mean 'ipsum'?  Vale.Spelling 

✖ 5 errors, 0 warnings and 0 suggestions in 3 files.
```

That's OK. But let's say I need to lint everything inside `content/` except for the `content/b/` subdir (because it is automatically generated content). Taking advantage of the `--glob` flag I'd try running `vale --glob='!content/b/*' ./content`. The result of that is this, though:

```
 content/a.md
 1:1  error  Did you really mean 'Lorem'?  Vale.Spelling 
 1:7  error  Did you really mean 'ipsum'?  Vale.Spelling 

✖ 2 errors, 0 warnings and 0 suggestions in 1 file.
```

`a.md` is linted (that's OK), `b/b.md` is not linted (which is also OK!), but... any other files **alphabetically after that** won't be linted (so no output for `c.md`).

If we rename `c.md` to `aa.md`, the result will be the expected one:

```
 content/a.md
 1:1  error  Did you really mean 'Lorem'?  Vale.Spelling 
 1:7  error  Did you really mean 'ipsum'?  Vale.Spelling 


 content/aa.md
 1:1  error  Did you really mean 'Lorem'?  Vale.Spelling 
 1:7  error  Did you really mean 'ipsum'?  Vale.Spelling 

✖ 4 errors, 0 warnings and 0 suggestions in 2 files.
```

This looks like a `vale` bug, doesn't it? Or is it a problem with https://github.com/gobwas/glob ? I've been playing around with the lib and seems to return the expected values in this case: https://play.golang.org/p/EBrkS4na90A ...

**Comment by jdkato** [comment]
Thanks for the report.

This does look like a bug on Vale's side. I'll try to fix this for the next release, but I'd really  like to re-write the entire implementation (as mentioned in #243).

**Comment by jdkato** [comment]
This should be fixed in v2.5.1.

**Comment by acalvo** [comment]
Works great, thanks! 🥳
